routes/controllers/web-api.js

```diff
@@ -1,23 +1,41 @@
 
+const {pick} = require('lodash');
+
 async function find(req, res) {
-    const {
-        query,
-        property,
-        author,
-        sort,
-        start,
-        amount,
-    } = req.query;
-
-    if (!query && !author && !sort) {
+    const properties = {
+        query: req.query.query,
+        property: req.query.property,
+        author: req.query.author,
+        sort: req.query.sort,
+        start: req.query.start,
+        amount: req.query.amount,
+    };
+
+    // Remove undefined properties
+    Object.keys(properties).forEach((key) => {
+        if (properties[key] === undefined) delete properties[key];
+    });
+
+    // Put a singular `property` into an array
+    if (properties.property !== undefined && !Array.isArray(properties.property)) {
+        properties.property = [properties.property];
+    }
+
+    // Reject invalid case
+    if (
+        properties.query === undefined &&
+        properties.author === undefined &&
+        properties.sort === undefined
+    ) {
         res.status(400).json({
             error: 'packagename is a required query parameter.',
         });
         return;
     }
 
-    // const matches = await req.dutyfree.searchForPackageByName(query);
-    res.json({ query, property, author, sort, start, amount });
+    const matches = await req.dutyfree.searchPackages(properties);
+
+    res.json(matches);
```

Comment:
So we can get better visibility on errors, we should wrap the body of this async function in a `try {} catch() {}` and send a `500` error back from the server if an error occurs.